### PR TITLE
perf: remove custom transport executor

### PR DESCRIPTION
**Pull request by olavloite**
Remove the custom executor provider that is set on the underlying generated clients and instead use the internal core gRPC executor provider. The latter is a shared executor provider for all gRPC channels that creates threads on demand. This prevents the creation of unnecessary threads at startup, and can reduce overall thread usage for applications that create multiple Spanner instances during their lifetime.

### CODE REVIEW

File: google-cloud-spanner/src/test/java/com/google/cloud/spanner/BaseSessionPoolTest.java [hunk 1]
```diff
@@ -54,7 +54,10 @@ public ScheduledExecutorService get() {
 
     @Override
     public void release(ScheduledExecutorService executor) {
-      executor.shutdown();
+      try {
```
Comment (olavloite) [inline google-cloud-spanner/src/test/java/com/google/cloud/spanner/BaseSessionPoolTest.java:57]: This cleans up errors that are being logged during tests, clogging up the test log.

File: google-cloud-spanner/src/test/java/com/google/cloud/spanner/SpannerOptionsTest.java [hunk 2]
```diff
@@ -923,14 +923,22 @@ public void testCustomAsyncExecutorProvider() {
   @Test
   public void testDefaultNumChannelsWithGrpcGcpExtensionEnabled() {
     SpannerOptions options =
-        SpannerOptions.newBuilder().setProjectId("test-project").enableGrpcGcpExtension().build();
+        SpannerOptions.newBuilder()
+            .setProjectId("test-project")
+            .setCredentials(NoCredentials.getInstance())
```
Comment (olavloite) [inline google-cloud-spanner/src/test/java/com/google/cloud/spanner/SpannerOptionsTest.java:928]: This cleans up errors that are being logged during tests, clogging up the test log.

File: google-cloud-spanner/src/test/java/com/google/cloud/spanner/SpannerOptionsTest.java [hunk 3]
```diff
@@ -923,14 +923,22 @@ public void testCustomAsyncExecutorProvider() {
...
     assertEquals(SpannerOptions.GRPC_GCP_ENABLED_DEFAULT_CHANNELS, options.getNumChannels());
   }
 
   @Test
   public void testDefaultNumChannelsWithGrpcGcpExtensionDisabled() {
-    SpannerOptions options = SpannerOptions.newBuilder().setProjectId("test-project").build();
+    SpannerOptions options =
+        SpannerOptions.newBuilder()
+            .setProjectId("test-project")
+            .setCredentials(NoCredentials.getInstance())
```
Comment (olavloite) [inline google-cloud-spanner/src/test/java/com/google/cloud/spanner/SpannerOptionsTest.java:940]: This cleans up errors that are being logged during tests, clogging up the test log.

File: google-cloud-spanner/src/test/java/com/google/cloud/spanner/SpannerOptionsTest.java [hunk 4]
```diff
@@ -943,6 +951,7 @@ public void testNumChannelsWithGrpcGcpExtensionEnabled() {
     SpannerOptions options1 =
         SpannerOptions.newBuilder()
             .setProjectId("test-project")
+            .setCredentials(NoCredentials.getInstance())
```
Comment (olavloite) [inline google-cloud-spanner/src/test/java/com/google/cloud/spanner/SpannerOptionsTest.java:954]: This cleans up errors that are being logged during tests, clogging up the test log.

File: google-cloud-spanner/src/test/java/com/google/cloud/spanner/SpannerOptionsTest.java [hunk 5]
```diff
@@ -954,6 +963,7 @@ public void testNumChannelsWithGrpcGcpExtensionEnabled() {
     SpannerOptions options2 =
         SpannerOptions.newBuilder()
             .setProjectId("test-project")
+            .setCredentials(NoCredentials.getInstance())
```
Comment (olavloite) [inline google-cloud-spanner/src/test/java/com/google/cloud/spanner/SpannerOptionsTest.java:966]: This cleans up errors that are being logged during tests, clogging up the test log.

File: google-cloud-spanner/src/test/java/com/google/cloud/spanner/SpannerOptionsTest.java [hunk 6]
```diff
@@ -972,12 +982,19 @@ public void checkCreatedInstanceWhenGrpcGcpExtensionDisabled() {
     Spanner spanner2 = options1.getService();
 
     assertNotSame(spanner1, spanner2);
+
+    spanner1.close();
```
Comment (olavloite) [inline google-cloud-spanner/src/test/java/com/google/cloud/spanner/SpannerOptionsTest.java:986]: This cleans up errors that are being logged during tests, clogging up the test log.

File: google-cloud-spanner/src/test/java/com/google/cloud/spanner/SpannerOptionsTest.java [hunk 7]
```diff
@@ -972,12 +982,19 @@ public void checkCreatedInstanceWhenGrpcGcpExtensionDisabled() {
...
+    spanner2.close();
   }
 
   @Test
   public void checkCreatedInstanceWhenGrpcGcpExtensionEnabled() {
     SpannerOptions options =
-        SpannerOptions.newBuilder().setProjectId("test-project").enableGrpcGcpExtension().build();
+        SpannerOptions.newBuilder()
+            .setProjectId("test-project")
+            .setCredentials(NoCredentials.getInstance())
```
Comment (olavloite) [inline google-cloud-spanner/src/test/java/com/google/cloud/spanner/SpannerOptionsTest.java:995]: This cleans up errors that are being logged during tests, clogging up the test log.

File: google-cloud-spanner/src/test/java/com/google/cloud/spanner/SpannerOptionsTest.java [hunk 8]
```diff
@@ -986,5 +1003,8 @@ public void checkCreatedInstanceWhenGrpcGcpExtensionEnabled() {
     Spanner spanner2 = options1.getService();
 
     assertNotSame(spanner1, spanner2);
+
+    spanner1.close();
```
Comment (olavloite) [inline google-cloud-spanner/src/test/java/com/google/cloud/spanner/SpannerOptionsTest.java:1007]: This cleans up errors that are being logged during tests, clogging up the test log.

**Review by thiagotnunes — APPROVED** [review]
LGTM

**Pull request merged**